Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/ArgumentSpace.java

```diff
@@ -0,0 +1,352 @@
+package nl.utwente.group10.ui.components.blocks.function;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import nl.utwente.group10.haskell.type.Type;
+import nl.utwente.group10.ui.components.ComponentLoader;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.value.ObservableValue;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.input.InputEvent;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.input.TouchEvent;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.Region;
+import javafx.scene.shape.Circle;
+
+/**
+ * Custom Pane that displays function's input and output and a knot to be able
+ * to control the amount of inputs that are to be applied (amount of currying).
+ * 
+ * The input arguments are ordered using the layoutX, and the knotIndex is based
+ * on this and the knot's layoutX. Usage of translateX is strictly for visual
+ * improvements.
+ * 
+ * Supports multi-touch interaction.
+ */
+public class ArgumentSpace extends Pane implements ComponentLoader{    
+    /** Horizontal space to keep between elements in ArgumentSpace. */
+    public static final double H_GAP = 10;
+    
+    /** Vertical height of the ArgumentSpace. */
+    public static final double HEIGHT = 50;
+    
+    /** The block to which this ArgumentSpace belongs. */
+    private FunctionBlock block;
+    
+    /** The knot, used to control the knot index. */
+    @FXML private Circle knot;
+    
+    /**
+     * The index of the knot, all inputs with index higher or equal to the
+     * knot are inactive, creating higher order functions.
+     */
+    private IntegerProperty knotIndex;
+    
+    /** List of input arguments. */
+    private List<InputArgument> leftArguments;
+    
+    /** The argument containing the function's output. */
+    @FXML private Label rightArgument;
+    
+    /** InputID of the current action (INPUT_ID_NONE if idle) */
+    private int inputID;
+    
+    /** InputID that represents no current inputID. */
+    public static final int INPUT_ID_NONE = -2;
+    
+    /** InputID that represents the mouse. */
+    public static final int INPUT_ID_MOUSE = -1;
+    
+    /**
+     * Constructs an ArgumentSpace belonging to the function block as given.
+     * @param block FunctionBlock to which this ArgumentSpace belongs.
+     */
+    public ArgumentSpace(FunctionBlock block, List<Type> inputSignatures) {
+        this.loadFXML("ArgumentSpace");
+        
+        this.block = block;
+        this.inputID = INPUT_ID_NONE;
+        leftArguments = new ArrayList<InputArgument>();
+        knotIndex = new SimpleIntegerProperty(0);
```

Comment:
Yes, for a Block that has not inputs, this is the only valid index. The FunctionBlock sets this to our default (size - 1) immediately after constructing the ArgumentSpace.